SHARD-2599 : enables local testing mode with high gas defaults

abdulazeem-tk4vr opened the pull request:
### **User description**
Adds a local testing mode that configures high gas defaults for local development environments.

This allows for faster development cycles by avoiding gas estimation on local networks, and instead using a constant gas amount. A warning is logged if local testing mode is enabled on a non-local network, to prevent accidental use in production environments.


___

### **PR Type**
Enhancement, Documentation


___

### **Description**
- Introduces local testing mode with high gas defaults

- Adds environment-based configuration for local testing

- Implements safety checks and warnings for misconfiguration

- Provides example config file for local testing setup


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>api.ts</strong><dd><code>Enable local testing mode for gas estimation logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/api.ts

<li>Implements local testing mode for gas estimation<br> <li> Uses constant gas value when local testing is enabled<br> <li> Emits log events and counts responses for local testing<br> <li> Comments out previous static gas estimate logic


</details>


  </td>
  <td><a href="https://github.com/shardeum/json-rpc-server/pull/171/files#diff-769911c416ccf8514d8fd941ae0abe8fb5c606ade0c218e22151a5f5f9f3d700">+11/-3</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>config.ts</strong><dd><code>Add localTesting config and environment validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/config.ts

<li>Adds <code>localTesting</code> config section with env-based toggles<br> <li> Sets default constant gas value for local testing<br> <li> Implements safety checks for local testing on non-local networks<br> <li> Logs warnings or confirmations based on environment


</details>


  </td>
  <td><a href="https://github.com/shardeum/json-rpc-server/pull/171/files#diff-c3095d5010e65c52737a98a5d618ea24049ebe90c8470752426081d70ed6e012">+29/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>local-testing.config.example</strong><dd><code>Provide example config for local testing mode</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

local-testing.config.example

<li>Adds example environment variable config for local testing<br> <li> Documents enabling local testing and customizing gas defaults<br> <li> Includes local network and port settings for development


</details>


  </td>
  <td><a href="https://github.com/shardeum/json-rpc-server/pull/171/files#diff-24f840960f4bec4c72bcd68ba4997ca8a2417471d7a137a189cf21dd61eb1a1f">+18/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

abdulazeem-tk4vr committed:
feat: enables local testing mode with high gas defaults

Adds a local testing mode that configures high gas defaults for local development environments.

This allows for faster development cycles by avoiding gas estimation on local networks, and instead using a constant gas amount. A warning is logged if local testing mode is enabled on a non-local network, to prevent accidental use in production environments.

abdulazeem-tk4vr commented:
do security review

mssabr01 commented on src/config.ts:
```diff
@@ -273,4 +278,28 @@ export const CONFIG: Config = {
   },
+  localTesting: {
+    enabled: Boolean(process.env.LOCAL_TESTING_MODE) || false,
+    highGasDefaults: Boolean(process.env.LOCAL_HIGH_GAS_DEFAULTS) || false,
```
Boolean() evaluates both "true" and "false" as true. 

Maybe change this to Boolean(process.env.LOCAL_TESTING_MODE === "true") and
Boolean(process.env.LOCAL_HIGH_GAS_DEFAULTS === "true")

abdulazeem-tk4vr replied:
I took reference from the existing bool variables on how the env variables should be assigned values
Looks like there are other places where such issues happen too
Should I fix them as well?

<img width="1461" height="800" alt="Image" src="https://github.com/user-attachments/assets/992d07c2-efe5-4865-9dcb-7932a1177d14" />

abdulazeem-tk4vr committed:
fix : enables static gas estimation and fixes env var parsing.

This commit improves gas estimation and configuration loading.

- Re-enables the static gas estimate feature.
- Corrects the parsing of boolean environment variables to ensure they are properly evaluated.
